src/js/common/schemaform/reducers/index.js

```diff
@@ -157,10 +159,22 @@ export default function createSchemaFormReducer(formConfig) {
         return _.set('loadedStatus', action.status, state);
       }
       case SET_IN_PROGRESS_FORM: {
-        const newState = _.set('loadedData', action.data, state);
+        let newState;
+
+        // if we're prefilling, we want to use whatever initial data the form has
+        if (action.prefilled) {
```

Comment:
Currently, prefilling blows away all data, when it should really be merging with our default data. In progress forms can just replace, because we're saving all the default data along with them.